### PR TITLE
debian dependencies

### PR DESCRIPTION
**Pull request by Alessandro-Barbieri**

### CODE REVIEW

File: rosdep/base.yaml [hunk 1]
```diff
@@ -349,9 +359,15 @@ checkinstall:
   debian: [checkinstall]
   ubuntu: [checkinstall]
 chromium-browser:
+  debian:
+    wheezy: [chromium-browser]
```
Comment (tfoote) [inline rosdep/base.yaml]: It looks like they've taken over the `chromium` key on debian. https://packages.debian.org/stretch/chromium

File: rosdep/base.yaml [hunk 2]
```diff
@@ -349,9 +359,15 @@ checkinstall:
...
   gentoo: [www-client/chromium]
   ubuntu: [chromium-browser]
 clang-format:
+  debian:
+    buster: [clang-format]
+    sid: [clang-format]
```
Comment (tfoote) [inline rosdep/base.yaml]: Please don't add sid rules. We don't set them since it's a moving target.

File: rosdep/base.yaml [hunk 3]
```diff
@@ -392,7 +413,10 @@ coinor-libosi-dev:
   ubuntu: [coinor-libosi-dev]
 collada-dom:
   arch: [collada-dom]
-  debian: [libcollada-dom2.4-dp-dev]
+  debian:
+    buster: [libcollada-dom2.4-dp-dev]
+    sid: [libcollada-dom2.4-dp-dev]
```
Comment (tfoote) [inline rosdep/base.yaml]: -sid

File: rosdep/base.yaml [hunk 4]
```diff
@@ -729,7 +766,10 @@ gawk:
   ubuntu: [gawk]
 gazebo:
   arch: [gazebo]
-  debian: [gazebo7]
+  debian:
+    buster: [gazebo7]
+    sid: [gazebo7]
```
Comment (tfoote) [inline rosdep/base.yaml]: -sid

File: rosdep/base.yaml [hunk 5]
```diff
@@ -750,7 +790,10 @@ gazebo5:
   ubuntu: [gazebo5]
 gazebo7:
   arch: [gazebo]
-  debian: [gazebo7]
+  debian:
+    buster: [gazebo7]
+    sid: [gazebo7]
```
Comment (tfoote) [inline rosdep/base.yaml]: -sid

File: rosdep/base.yaml [hunk 6]
```diff
@@ -887,6 +937,11 @@ gpsd:
   ubuntu: [gpsd]
 gradle:
   arch: [gradle]
+  debian:
+    buster: [gradle]
+    jessie: [gradle]
+    sid: [gradle]
```
Comment (tfoote) [inline rosdep/base.yaml]: gradle appears to be available on all versions: https://packages.debian.org/stretch/gradle

File: rosdep/base.yaml [hunk 7]
```diff
@@ -1113,7 +1176,10 @@ jasper:
   ubuntu: [libjasper-dev]
 java:
   arch: [jdk7-openjdk]
-  debian: [openjdk-7-jdk]
+  debian:
+    jessie: [openjdk-7-jdk]
+    sid: [openjdk-7-jdk]
```
Comment (tfoote) [inline rosdep/base.yaml]: -sid

File: rosdep/base.yaml [hunk 8]
```diff
@@ -1316,17 +1391,26 @@ libcegui-mk2-dev:
   gentoo: [dev-games/cegui]
   ubuntu: [libcegui-mk2-dev]
 libceres-dev:
+  debian:
+    buster: [libceres-dev]
+    sid: [libceres-dev]
```
Comment (tfoote) [inline rosdep/base.yaml]: -sid

File: rosdep/base.yaml [hunk 9]
```diff
@@ -1316,17 +1391,26 @@ libcegui-mk2-dev:
...
+    wheezy: [libcoin60-dev]
   fedora: [Coin2-devel]
   gentoo: [media-libs/coin]
   ubuntu: [libcoin60-dev]
 libcoin80-dev:
-  debian: [libcoin80-dev]
+  debian:
+    buster: [libcoin80-dev]
+    jessie: [libcoin80-dev]
+    sid: [libcoin80-dev]
```
Comment (tfoote) [inline rosdep/base.yaml]: -sid

File: rosdep/base.yaml [hunk 10]
```diff
@@ -1418,7 +1504,9 @@ libexpat1-dev:
 libfcl-dev:
   arch: [fcl]
   debian:
+    buster: [libfcl-dev]
     jessie: [libfcl-0.5-dev]
+    sid: [libfcl-dev]
```
Comment (tfoote) [inline rosdep/base.yaml]: -sid

File: rosdep/base.yaml [hunk 11]
```diff
@@ -1514,8 +1605,11 @@ libftdi-dev:
   ubuntu: [libftdi-dev]
 libftdipp-dev:
   debian:
+    buster: [libftdipp1-dev]
     jessie: [libftdipp-dev]
+    sid: [libftdipp1-dev]
```
Comment (tfoote) [inline rosdep/base.yaml]: -sid

File: rosdep/base.yaml [hunk 12]
```diff
@@ -1538,7 +1632,10 @@ libgazebo5-dev:
   ubuntu: [libgazebo5-dev]
 libgazebo7-dev:
   arch: [gazebo]
-  debian: [libgazebo7-dev]
+  debian:
+    buster: [libgazebo7-dev]
+    sid: [libgazebo7-dev]
```
Comment (tfoote) [inline rosdep/base.yaml]: -sid

**Review by tfoote — CHANGES_REQUESTED** [review]
Overall this looks good. I had a few comments, and pointed out a bunch of places with sid keys. We don't target sid because it's a rolling release that changes over time. Now that buster has been announced we can begin to converge on that for new keys.

**Comment by Alessandro-Barbieri** [comment]
finished

**Review by tfoote — APPROVED** [review]

**Pull request merged**